spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/sampler/UuidToBigIntegerConverter.java

```diff
@@ -0,0 +1,23 @@
+package org.springframework.cloud.sleuth.sampler;
+
+import java.math.BigInteger;
+import java.util.UUID;
+
+class UuidToBigIntegerConverter {
+
+	/**
+	 * Takes two sets of bits (most and least significant ones), puts them together
+	 * and represents as a BigInteger
+	 *
+	 * @param uuid - UUID to convert
+	 * @return - BigInteger representation of the UUID
+	 */
+	static BigInteger uuidToBigInt(UUID uuid) {
```

Comment:
said earlier, I'd love this to be the decoder function from String -> BigInteger. The UUID type is a convenience for that conversion.